### PR TITLE
Timeout forkserver

### PR DESCRIPTION
**Pull request by tokatoka**
#82 #111 
I've made the TimeoutForkserverExecutor that wraps around the normal ForkserverExecutor and can check timeout also. It can be used similarly as the TimeoutExecutor, (just pass the Executor and the Duration to call the constructor).

### CODE REVIEW

**Comment by tokatoka** [comment]
I have to fix some clippy errors

**Comment by tokatoka** [comment]
I don't like it when clippy complains "XXX is being shadowed" 😓

File: libafl/src/executors/forkserver.rs [hunk 1]
```diff
@@ -240,11 +260,143 @@ impl Forkserver {
...
-    pub fn write_ctl(&mut self, val: i32) -> Result<usize, io::Error> {
+    pub fn write_ctl(&mut self, val: i32) -> Result<usize, Error> {
         let slen = self.ctl_pipe.write(&val.to_ne_bytes())?;
 
         Ok(slen)
     }
+
+    pub fn read_st_timed(&mut self, timeout: &mut TimeVal) -> Result<Option<(usize, i32)>, Error> {
+        let mut buf: [u8; 4] = [0u8; 4];
+        let st_read = self.st_pipe.read_end().unwrap();
```
Comment (domenukk) [inline libafl/src/executors/forkserver.rs:271]: Maybe use `?` instead of `unwrap` here?

File: libafl/src/executors/forkserver.rs [hunk 2]
```diff
@@ -240,11 +260,143 @@ impl Forkserver {
...
+        // We'll pass a copied timeout to keep the original timeout intact, because select updates timeout to indicate how much time was left. See select(2)
+        let sret = select(
+            Some(readfds.highest().unwrap() + 1),
+            &mut readfds,
+            None,
+            None,
+            &mut copy,
+        )?;
+        if sret > 0 {
+            let rlen = self.st_pipe.read(&mut buf)?;
```
Comment (domenukk) [inline libafl/src/executors/forkserver.rs:284]: `read_exact`?

**Comment by domenukk** [comment]
Looks good, awesome stuff! :)

**Review by domenukk — APPROVED** [review]

**Comment by domenukk** [comment]
Should we maybe change the forkserver example to use the TimeoutForkserver now?

**Comment by tokatoka** [comment]
ok

**Pull request merged**